### PR TITLE
Fix and optimize MSVC flags

### PR DESCRIPTION
**Pull request by daschuer**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -259,6 +266,11 @@ if(MSVC)
       string(REPLACE "/RTC1" "" CMAKE_CXX_FLAGS_DEBUG  "${CMAKE_CXX_FLAGS_DEBUG}")
       string(REPLACE "/RTC1" "" CMAKE_C_FLAGS_DEBUG  "${CMAKE_C_FLAGS_DEBUG}")
     elseif(CMAKE_BUILD_TYPE STREQUAL "RelWithDebInfo")
+      # For some reasons cmake uses /Ob1 in RelWithDebInfo https://gitlab.kitware.com/cmake/cmake/-/issues/20812
+      # /O2 this implies /Od2
```
Comment (JoergAtGithub) [inline CMakeLists.txt:270]:
```suggestion
```
I guess this line is a copy/paste failure?

Comment (daschuer) [reply CMakeLists.txt:270]: No, but you are right that this is not understandable. I'll try to improve it.

**Comment by daschuer** [comment]
Done

**Comment by daschuer** [comment]
@JoergAtGithub can this be merged for 2.3.5?

**Comment by JoergAtGithub** [comment]
The changes looking good to me, but I haven't tested it yet.

**Comment by JoergAtGithub** [comment]
I built this branch as x64_portable and x64_off. Both worked on showed the expected debugging behavior. Thank you!

**Pull request merged**